### DISCUSSION TITLE
How can I make a widget inside a scroll resize when the window is resized?

### DISCUSSION THREAD

**Discussion by LaBatata101** [fltk-rs/fltk-rs#899] [Q&A] [answered]
```rust
fn main() {
    let app = app::App::default().with_scheme(app::Scheme::Gtk);

    let mut win = window::Window::default()
        .with_size(800, 600)
        .with_label("hello world");

    let mut scroll = group::Scroll::default().size_of_parent();
    
    let mut frame = Frame::default().with_label("TEST RESIZE").size_of_parent();
    frame.set_frame(FrameType::BorderBox);
    frame.set_color(Color::BackGround | Color::Red);

    scroll.end();

    scroll.set_frame(FrameType::RoundedBox);
    scroll.set_type(group::ScrollType::Vertical);
    scroll.make_resizable(true);

    win.make_resizable(true);

    win.end();
    win.show();

    app.run().unwrap();
}
```

Why in the code above when I resize the window, the `frame` inside `Scroll` doesn't resize?
![image](https://user-images.githubusercontent.com/20308796/134256350-4ec768e5-e39f-48cb-b396-90382933c1f1.png)

**Comment by MoAlyousef** [comment, marked as answer]
That's the default behavior of the Scroll widget in regards to its children. To override it you can either override the draw() or handle() methods of either frame or window.

Since I'm not sure what you're trying to do, the following might give an idea:
```rust
    frame.draw(move |f| {
        f.resize(0, 0, win.w(), win.h());
    });
    // Or:
    frame.handle(move |f, ev| match ev {
        Event::Resize => {
            f.resize(0, 0, win.w(), win.h());
            true
        }
        _ => false,
    });
```
Or for the window:
```rust
    win.draw(move |w| {
        frame.resize(0, 0, w.w(), w.h());
    });
    // Or:
    win.handle(move |w, ev| match ev {
        Event::Resize => {
            frame.resize(0, 0, w.w(), w.h());
            true
        }
        _ => false,
    });
```

**Reply by LaBatata101** [reply]
Thanks for the quick response. The second code example for the `frame` solved my problem. 

But in the docs for the `Resize` event it says `Window Resize Event. Avoid resizing the widget during a resize event to avoid infinite recursion` should I be worried about that? Or do you know how often this infinite recursion may happen?

**Reply by MoAlyousef** [reply]
No problem. I'll change the docs regarding that phrase. The idea is that resizing the widget's parent in the Resize event will also attempt to resize the children thus re-triggering the Resize event again and again. This applies to parents which actually resize their children widgets, such as windows and packs.
